### ISSUE TITLE
s.instance_variable_get(:@pool) returning nil

### ISSUE THREAD

**Issue by gmoore** [nbudin/devise_cas_authenticatable#101]
Trying to use single sign out.  

Here's the log on the app side when someone signs out:

```
Started POST "/users/service" for 127.0.0.1 at 2015-04-27 09:52:16 -0400
Processing by Devise::CasSessionsController#single_sign_out as */*
  Parameters: {"logoutRequest"=>"<samlp:LogoutRequest xmlns:samlp=\"urn:oasis:names:tc:SAML:2.0:protocol\" xmlns:saml=\"urn:oasis:names:tc:SAML:2.0:assertion\" ID=\"da185460-0f04-4a2b-8d6f-978ccb403259\" Version=\"2.0\" IssueInstant=\"2015-04-27 09:52:16 -0400\">\n  <saml:NameID>@NOT_USED@</saml:NameID>\n  <samlp:SessionIndex>ST-14301427174765-MXXEHFrbaM3KMiYP92RktD18V4bF3nVRLjNQ6uYR</samlp:SessionIndex>\n</samlp:LogoutRequest>\n"}
Intercepted single-sign-out request for CAS session ST-14301427174765-MXXEHFrbaM3KMiYP92RktD18V4bF3nVRLjNQ6uYR.
Found session id 01e326f55b3e9d942b1957f7b9fa9540 for index ST-14301427174765-MXXEHFrbaM3KMiYP92RktD18V4bF3nVRLjNQ6uYR
Found Session ID 01e326f55b3e9d942b1957f7b9fa9540 with index key ST-14301427174765-MXXEHFrbaM3KMiYP92RktD18V4bF3nVRLjNQ6uYR
Single Sign Out from session store: RedisSessionStore
Completed 500 Internal Server Error in 3.6ms
** [Airbrake] Notice was not sent due to configuration:         
  Environment Monitored? false         
  API key set? true

NoMethodError - undefined method `del' for nil:NilClass:
  devise_cas_authenticatable (1.3.7) lib/devise_cas_authenticatable/single_sign_out.rb:47:in `destroy_session_by_id'
  devise_cas_authenticatable (1.3.7) app/controllers/devise/cas_sessions_controller.rb:67:in `destroy_cas_session'
  devise_cas_authenticatable (1.3.7) app/controllers/devise/cas_sessions_controller.rb:41:in `single_sign_out'
```

The line that's failing is 

    current_session_store.instance_variable_get(:@pool).del(sid)

The problem is that `current_session_store.instance_variable_get(:@pool)` is returning nil

    Loading development environment (Rails 3.2.21)
    [1] pry(main)> s = Rails.application.config.session_store 
    => ActionDispatch::Session::RedisStore
    [2] pry(main)> s.instance_variable_get(:@pool)

I've tried this with both redis-rails and redis-session-store gems

My session config:

```
  config.session_store :redis_store, 
          servers: { :db => 0,
                     :namespace => "myapp:session:",
                     :expires_in => 30.days }
```

Redis is up. I can connect to it though `redis-cli` and see that the keys are being added when sessions are created.

I can't find docs on what "pool" is supposed to be or be set to. Is there something I'm missing?

Thanks.

**Comment by gmoore** [comment]
Of course it's working now. I rebooted everything and deleted all the cookies and gave it a go and the first try worked. I'll re-open this is the problem comes up again.

**Comment by ledestin** [comment]
@nbudin Hi, I have a similar problem on Rails 5. Probably, the gems used have changed a bit:
1. `@pool` was moved inside of `@conn`. At least that's where it is now.
2. `connection_pool` doesn't work with `set`, etc. It needs `with { |redis| redis.set }`.

I'm using `redis-rails` and `connection_pool`.

Would you be interested in a PR? I've successfully patched your gem locally.

**Comment by nbudin** [comment]
@ledestin Yes, I'd love a PR.  Thank you!

**Comment by ledestin** [comment]
https://github.com/nbudin/devise_cas_authenticatable/pull/150

**Comment by nbudin** [comment]
This is now fixed in v1.10.4.  Thanks @ledestin for the PR!
